### PR TITLE
[5.9] Update Vue to v2.7, VueCLI to v5, Node to v18, NPM to v9

### PR DESCRIPTION
**Pull request by dobromir-hristov**
- **Explanation:** Updates major prod and dev deps, like Vue, VueCLI, Node to latest stable and supported versions. Removes critical dep issues.
- **Scope:** Entire app, as the dev-tools got updated
- **Issue:** rdar://100963792
- **Risk:** High, updates dev dependencies and Vue version, with slight css changes
- **Testing:** Manually tested app in lots of scenarios
- **Reviewer:** @marinaaisa, @hqhhuang, @mportiz08 
- **Original PR:** #440

### CODE REVIEW

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Comment by mportiz08** [comment]
I would personally consider this as potentially "High" risk simply due to the number of files affected and the very wide impact, although everything seems to working well from my testing.

File: src/components/CallToActionButton.vue [hunk 1]
```diff
@@ -9,9 +9,8 @@
 -->
 
 <template>
-  <DestinationDataProvider v-if="action" :destination="action">
+  <DestinationDataProvider v-if="action" :destination="action" v-slot="{ url, title }">
```
Comment (eango) [inline src/components/CallToActionButton.vue:12]: Has the Call to Action been tested with this change? I recall we had an issue with this in the past

Comment (dobromir-hristov) [reply src/components/CallToActionButton.vue:12]: This is just a syntax change for the slot. The tests would have failed if its wrong :)

**Comment by franklinsch** [comment]
@dobromir-hristov Can you elaborate on the benefit of including this in the 5.9 release? Is it worth the risk you've outlined?

**Comment by dobromir-hristov** [comment]
We wanted to get this out of the door as soon as possible, as docc-render has lots of critical dependency vulnerabilities that we cannot fix in other ways. 

Ofc if you think its risky to do it now, we can wait for a more appropriate time.

**Comment by mportiz08** [comment]
We discussed this and feel that it is probably too risky to take for 5.9 due to the large number of changes and limited end-user benefit.

We will continue with the updated dependencies on main going forward though.

**Pull request closed without merging**